frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -900,11 +947,19 @@ pub mod pallet {
             reward_destination: RewardDestination,
         ) -> DispatchResultWithPostInfo {
             Self::ensure_pallet_enabled()?;
-            let staker = ensure_signed(origin)?;
+            let staker = ensure_signed(origin.clone())?;
             let mut ledger = Self::ledger(&staker);
 
             ensure!(!ledger.is_empty(), Error::<T>::NotActiveStaker);
 
+            ensure!(reward_destination != ledger.reward_destination, Error::<T>::SameRewardDestinationNotAllowed);
+
+            // if current reward_destionation is RewardDestination::DelegateBalance and staker wants tot change it so have to remove 
+            // delegate info from mapping (for safety) (optional feature)
+            if ledger.reward_destination == RewardDestination::DelegateBalance {
```

Comment:
What also needs to be updated for `set_reward_destination` call after this change?

Reply:
Not sure, AFAIK there are only 2 storage items that concerns with RewardDestination.
1. Ledger which gets updated at the end of this function.
2. DelegateInfo which gets updated in `remove_all_delegatee_for_delegator` function.

Only thing that I can think of is benchmarks. This function now has an additional if statement which needs to be accounted for in benchmark.

Kindly let me know if I missed something.